src/braket/devices/local_simulator.py

```diff
@@ -204,9 +204,13 @@ def run_batch(  # noqa: C901
                         f"{unbounded_parameters}"
                     )
 
-        with Pool(min(max_parallel, len(tasks_and_inputs))) as pool:
-            param_list = [(task, shots, inp, *args, *kwargs) for task, inp in tasks_and_inputs]
-            results = pool.starmap(self._run_internal_wrap, param_list)
+        if isinstance(simulator := self._delegate, MultiSimulator):
+            all_tasks, all_inputs = tuple(zip(*tasks_and_inputs))
+            results = simulator.run_multiple(all_tasks, shots, inputs=all_inputs, *args, **kwargs)
```

Comment:
Note - calling `simulator.run_multiple()` directly here isn't correct. We need to delegate this through `self._run_internal` so that the result gets converted into the right type of object.